include/client/open.inc.php

```diff
@@ -9,6 +9,9 @@
 
 $info=($_POST && $errors)?Format::htmlchars($_POST):$info;
 
+if(array_key_exists('topicId',$_GET) && preg_match('/^\d+$/',$_GET['topicId']))
+    $info['topicId'] = intval($_GET['topicId']);
+
```

Comment:
I think you should move this code in the block below:
```php
if (!$info['topicId']) {
    if (array_key_exists('topicId',$_GET) && preg_match('/^\d+$/',$_GET['topicId']))
        $info['topicId'] = intval($_GET['topicId']);
    else
        $info['topicId'] = $cfg->getDefaultTopicId();
}
```
You could also perform a stronger check to ensure valid topic id:
```php
if (array_key_exists('topicId',$_GET) && Topic::lookup($_GET['topicId']))
```